core/admin/mailu/ui/views/languages.py

```diff
@@ -2,8 +2,8 @@
 
 import flask
 
-
-@ui.route('/language/<language>', methods=['GET'])
+@ui.route('/language/<language>', methods=['POST'])
 def set_language(language=None):
     flask.session['language'] = language
-    return flask.redirect(flask.url_for('.user_settings'))
```

Comment:
Hmm, you should keep a redirect here; the idiom is POST+redirect (302) ... so that if the user refreshes the page he doesn't get shown the scary "do you want to resubmit the form" dialog.

Reply:
The language "POST" ist not called via a form submit but by an ajax call.
So the user won't see resubmit dialogs etc - the call only needs a "200 OK" and triggers a reload of the current page (see app.js).
`
$.post({
            url: $(this).attr('href'),
            success: function() {
                location.reload();
            },
        });
`

But we could discuss about an error handler or a warning about losing form data when changing the language while configuring something.

Reply:
OK the important thing is that there is a GET following... I guess you could replace the success: by a .done() afterwards instead